src/components/Button.tsx

```diff
@@ -0,0 +1,79 @@
+import {StyleSheet, Text, TouchableOpacity} from "react-native";
+import React from "react";
+import {Ionicons} from "@expo/vector-icons";
+
+interface ButtonProps {
+    pressHandler?: () => void,
+    text?: string,
+    icon?: string,
+    styles?: {
+        button?: Record<string, any>,
+        text?: Record<string, any>,
+        icon?: Record<string, any>
+    }
+    type?: ButtonType,
+    size?: ButtonSize
+}
+
+export enum ButtonType {
+    primary,
+    secondary,
+    clean,
+    transparent
+}
+
+export enum ButtonSize {
+    sm,
+    md,
+    xl
+}
+
+const Button = (props: ButtonProps) => {
+    const {
+        pressHandler = () => {
```

Comment:
If you make the component that replaces the default one, usually it's cool to keep consistent naming. For example, every touchable element in RN has property `onPress`. To not confuse developers (and make it more intuitive), it's better to call your prop this way as well.